geoportailv3/static/js/locationControl.js

```diff
@@ -0,0 +1,166 @@
+/**
+ * @fileoverview This file defines the geolocation control.
+ *
+ */
+
+goog.require('ol.Feature');
+goog.require('ol.FeatureOverlay');
+goog.require('ol.Geolocation');
+goog.require('ol.control.Control');
+
+
+goog.provide('app.LocationControl');
+
+
+
+/**
+ * @constructor
+ * @extends {ol.control.Control}
+ * @param {Object=} opt_options Control options.
+ */
+app.LocationControl = function(opt_options) {
+
+  var options = opt_options || {};
```

Comment:
I'd suggest using `goog.isDef`:

```js
var options = goog.isDef(opt_options) ? opt_options : {};
```

And maybe you don't need that at all, because the control is always passed options in the application.